src/solvers/refinement/string_constraint_generator_testing.cpp

```diff
@@ -64,11 +62,10 @@ exprt string_constraint_generatort::add_axioms_for_is_prefix(
     const exprt witness = fresh_exist_index("witness_not_isprefix", index_type);
     const exprt strings_differ_at_witness = and_exprt(
       axiom_for_is_positive_index(witness),
-      and_exprt(
-        prefix.axiom_for_length_gt(witness),
-        notequal_exprt(str[plus_exprt(witness, offset)], prefix[witness])));
+      prefix.axiom_for_length_gt(witness),
+      notequal_exprt(str[plus_exprt(witness, offset)], prefix[witness]));
     const exprt s1_does_not_start_with_s0 = or_exprt(
-      not_exprt(positive_offset),
+      not_exprt(offset_within_bounds),
```

Comment:
You are changing something that you have only introduced in this PR, if I got this right. If so, you might want to apply some squashing.